### ISSUE TITLE
Cool stuff ... but how can I use it?

### ISSUE THREAD

**Issue by shuwang1** [acebrianjuan/gnss-sdr-monitor#6]
I think this looks really cool!  However, how can I use it?  

Does it only need to listen to the RTCM reports?  Say ... I have a GNSS measurement engine, which output RTCM report on  192.168.1.1/1234.  Then, I just add this IP info to GNSS-SDR monitor and that is it, right?

**Comment by acebrianjuan** [comment]
Hi Shu Wang,

This GUI is designed to receive real-time data from GNSS-SDR over the network either **locally** (GNSS-SDR and GUI instances running on the same machine) or **remotely** (GNSS-SDR and GUI instances running on different machines).

![gui-modes](https://user-images.githubusercontent.com/34104100/73607167-098e3a80-45b3-11ea-9ffc-7c509da325dd.png)

The data that is required by the GUI to work comes from GNSS-SDR via two serialization streams over UDP: the [**Monitor block streaming**](https://gnss-sdr.org/docs/sp-blocks/monitor/) and the [**PVT block streaming**](https://gnss-sdr.org/docs/sp-blocks/pvt/#custom-streaming), which are independent from each other and provide different kinds of information. In addition, the GUI provides a widget for communicating with the receiver interactive interface ([Telecommand via TCP/IP](https://gnss-sdr.org/docs/sp-blocks/global-parameters/#telecommand-via-tcpip)) by allowing you to send special commands to GNSS-SDR.

![GeneralBlockDiagram](https://user-images.githubusercontent.com/34104100/73399184-16770980-42e7-11ea-8f21-080d4f17244c.png)
<sup>Figure: GNSS-SDR general block diagram from the documentation page: <a href="https://gnss-sdr.org/docs/sp-blocks/">https://gnss-sdr.org/docs/sp-blocks/</a>. The Monitor block streaming and PVT block streaming are highlighted in red. The Telecommand module is highlighted in green. </sup>

Therefore, if you want to use the GUI, it is essential that _at least_ you **activate both serialization data streams in your GNSS-SDR receiver configuration file** by following these instructions:

In your configuration file, append the following lines to the `#PVT CONFIG#` section. Put the IP address of the machine where the GUI is running and choose a port number at your own discretion. I am using the loopback address and port 1111 as an example.

```
PVT.enable_monitor=true
PVT.monitor_client_addresses=127.0.0.1
PVT.monitor_udp_port=1111
```

This will activate the **serialized stream of PVT data** to the GUI, which will let you see the position reported by the receiver on a live map as well as a charts of the reported altitude and dilution of precision.

Next, add the following chunk in the `#MONITOR CONFIG#` section. Again, choose the IP address of the machine where the GUI is running and a port number that work best for you.

```
Monitor.enable_monitor=true
Monitor.decimation_factor=1
Monitor.client_addresses=127.0.0.1
Monitor.udp_port=1112
```

This will activate the **serialized stream of data from the Monitor block** to the GUI, which will let you see a table-like widget with the status of each channel accompanied by data readings of the acquisition and tracking stages.

Making sure that the GUI setup matches that of the GNSS-SDR configuration file is a critical step for the connection to work. In the menu bar of the GUI go to `Edit > Preferences`, and verify that the port numbers are correct.

<p align="center">
  <img src="https://user-images.githubusercontent.com/34104100/73596584-8ffb3b80-4523-11ea-8322-6c1f42544ca5.png"/>
</p>

Finally, if you want to make use of the **Telecommand** functionality (which is something I would recommend), append the following lines to the `#GLOBAL OPTIONS#` section of the configuration file.

```
GNSS-SDR.telecommand_enabled=true
GNSS-SDR.telecommand_tcp_port=3333
```

And then enter the IP address and TCP port number in the widget.

![telecommand-widget](https://user-images.githubusercontent.com/34104100/73596686-a950b780-4524-11ea-8873-1fb1e809efc8.png)

This will let you send commands to control the GNSS-SDR instance that you are monitoring with the GUI.

Once you complete these steps you are all set.

I hope this information is useful for you.
Cheers,
Álvaro

**Comment by acebrianjuan** [comment]
I have added the instructions from my [comment](https://github.com/acebrianjuan/gnss-sdr-monitor/issues/6#issuecomment-581124725) to the README file in commit https://github.com/acebrianjuan/gnss-sdr-monitor/commit/79ff0e2bac36aec153c31df08278807273e66cdb

**Comment by dengxu3** [comment]
I can't help thinking that are the preference default values of GNSS_Synchro port and Moniter_pvt port opposite？
I mean it should 1111 for Moniter_Pvt port value. Am I right?  It's kind of confusing.

**Comment by acebrianjuan** [comment]
Hi @dengxu3,

> I can't help thinking that are the preference default values of GNSS_Synchro port and Moniter_pvt port opposite？

Yes, that is correct. I mixed up the port numbers in the instructions by mistake.

> I mean it should 1111 for Moniter_Pvt port value. Am I right? It's kind of confusing.

Yes, you are right.

I have just pushed a quick fix in https://github.com/acebrianjuan/gnss-sdr-monitor/commit/55037f5dcadebac249834efb3b3dca10b84a2d84 updating the screenshot of the _Preferences_ window.
However, I probably need to rework the instructions in the README file to make them more clear. In addition, given that, as of late, a number of users (including yourself) have reported being confused by the port names I will address issue https://github.com/acebrianjuan/gnss-sdr-monitor/issues/11 ASAP.

Thank you for reporting the error.

Regards,
Álvaro
